### PR TITLE
chore(transformer-directives): deconstruct index.ts

### PR DESCRIPTION
**Pull request by chris-zhu**
Deconstruct the function to make it easier to read. If you don't like it, please close pr.

This PR change: 
- deconstruct logic

### CODE REVIEW

File: packages/transformer-directives/src/apply.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+type Writeable<T> = { -readonly [P in keyof T]: T[P] }
+
+interface ApplyContext {
+  offset?: number
+  code: MagicString
+  uno: UnoGenerator
+  childNode: CssNode
+  node: Rule
+}
```
Comment (hannoeru) [inline packages/transformer-directives/src/apply.ts:16]: I think we can combine these three contexts into one context. and in `ApplyContext` just put other params into function arguments.

**Comment by antfu** [comment]
@hannoeru feel free to merge once it looks good to you

File: packages/transformer-directives/src/index.ts [hunk 2]
```diff
@@ -24,6 +24,18 @@ export interface TransformerDirectivesOptions {
    * @default true
    */
   throwOnMissing?: boolean
+  /**
+   * Overwrite code with offset
+   *
+   * @default 0
+   */
+  offset?: number
```
[the remark this answers is not in the document]
Comment (chris-zhu) [reply packages/transformer-directives/src/index.ts]: I'm not sure if `offset` is used in other directives. Now I have it reseted.

File: packages/transformer-directives/src/index.ts [hunk 3]
```diff
@@ -69,212 +74,20 @@ export async function transformDirectives(
...
-      await Promise.all(
-        node.block.children.map(async (childNode, _childItem) => {
-          if (childNode.type === 'Raw')
-            return transformDirectives(code, uno, options, filename, childNode.value, calcOffset(childNode.loc!.start.offset))
-          await handleApply(node, childNode)
-        }).toArray(),
-      )
-    }
+    if (isApply && node.type === 'Rule')
+      await handleApply(options, { uno, code, node }, filename)
```
[the remark this answers is not in the document]
Comment (chris-zhu) [reply packages/transformer-directives/src/index.ts]: Well, it's up to you. I think this context is too complicated, some properties are custom for `apply` and it becomes bloated

**Review by hannoeru — APPROVED** [review]

**Pull request merged**